crates/flow-client/src/client.rs

```diff
@@ -331,10 +338,15 @@ pub async fn refresh_authorizations(
                 access_token: String,
                 refresh_token: Option<RefreshToken>, // Set iff the token was single-use.
             }
+            // We either never had an access token, or we had one and it expired,
+            // in which case the client may have an invalid access token configured.
+            // The `generate_access_token` RPC only needs the provided refresh token
+            // for authentication, so we should use an unauthenticated client to make
+            // the request.
             let Response {
                 access_token,
                 refresh_token: next_refresh_token,
-            } = api_exec::<Response>(client.rpc(
+            } = api_exec::<Response>(client.clone().as_anonymous().rpc(
```

Comment:
nit: couldn't this just be `with_creds(None)` ?

I also think that method should be renamed `with_user_access_token()` to match `new()` and the field itself...

Reply:
> I also think that method should be renamed with_user_access_token() to match new() and the field itself...

Yup, now that the client doesn't have a refresh token, that sounds right to me

> nit: couldn't this just be with_creds(None) ?

Yes. I don't remember why `with_creds` does `user_access_token: user_access_token.or(self.user_access_token)`, but none of the usages of it currently pass anything other than `Some(..)`, so I think it's safe to make it simply return 
```rust
        Self {
            user_access_token,
            ..self
        }
```